pkg/util/containers/v2/metrics/containerd/collector.go

```diff
@@ -143,10 +149,52 @@ func (c *containerdCollector) GetContainerNetworkStats(containerID string, cache
 	}
 }
 
+// GetContainerIDForPID returns the container ID for given PID
+func (c *containerdCollector) GetContainerIDForPID(pid int, cacheValidity time.Duration) (string, error) {
+	currentTime := time.Now()
+	strPid := strconv.Itoa(pid)
+
+	cID, found, _ := c.pidCache.Get(currentTime, strPid, cacheValidity)
+	if found {
+		return cID.(string), nil
+	}
+
+	// If we've done a full refresh within cacheValidity, we do not trigger another full refresh
+	_, found, _ = c.pidCache.Get(currentTime, pidCacheFullRefreshKey, cacheValidity)
+	if found {
+		return "", nil
+	}
+
+	// Full refresh
+	containers, err := c.client.Containers()
+	if err != nil {
+		return "", err
+	}
+
+	c.pidCache.Store(currentTime, pidCacheFullRefreshKey, struct{}{}, nil)
+	for _, container := range containers {
+		_, _ = c.getPIDs(currentTime, container)
```

Comment:
Not sure if I like the design of the `getPIDs` func. It returns errors but they're never handled, and it does store things in the cache, but does not ever read from it. I suggest extracting the cache logic from there and putting it inline here, and also handling the errors (just logging is fine I think, even if it's just debug level).

Reply:
I think I planned to re-use the function at some point but it's not the case in the end, so yes I can simplify